lightning-background-processor/src/lib.rs

```diff
@@ -1171,4 +1209,164 @@ mod tests {
 		let bg_processor = BackgroundProcessor::start(persister, event_handler, nodes[0].chain_monitor.clone(), nodes[0].node.clone(), nodes[0].no_gossip_sync(), nodes[0].peer_manager.clone(), nodes[0].logger.clone(), Some(nodes[0].scorer.clone()));
 		assert!(bg_processor.stop().is_ok());
 	}
+
+	#[test]
+	fn test_payment_path_scoring() {
```

Comment:
Hmm... seems this test could be *much* simpler if we could just inject an event in `ChannelManager` lol. We're really just testing that a method is called when given an event. How the event is produced isn't all that importatnt.

Reply:
I agree lol. I don't think `BackgroundProcessor` can use `cfg(test)` methods on `ChannelManager`, though. Think it's fine to expose a `cfg(feature = _test_utils)` method to push an event?

Reply:
I'd be in favor of something like that unless there's a compelling reason to do otherwise.